docs/topics/collection-grouping.md

```diff
@@ -2,22 +2,33 @@
 
 The Kotlin standard library provides extension functions for grouping collection elements.
 The basic function [`groupBy()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/group-by.html) takes a
-lambda function and returns a `Map`. In this map, each key is the lambda result and the corresponding value is the `List`
+lambda function and returns a `Map`. In this map, each key is the lambda result, and the corresponding value is the `List`
 of elements on which this result is returned. This function can be used, for example, to group a list of `String`s by
 their first letter. 
 
 You can also call `groupBy()` with a second lambda argument – a value transformation function.
 In the result map of `groupBy()` with two lambdas, the keys produced by `keySelector` function are mapped to the results
 of the value transformation function instead of the original elements.
 
-```kotlin
+This example illustrates using the `for` operator to iterate through the groups created by the `groupBy()` function on the resulting `Map`:
```

Comment:
I would suggest listing the actions in the same order as they appear in the code snippet: grouping by first letter, transforming to uppercase, and using `for`.